readthedocs/restapi/views/integrations.py

```diff
@@ -60,6 +65,8 @@ def post(self, request, project_slug):
         except Project.DoesNotExist:
             raise NotFound('Project not found')
         self.data = self.get_data()
+        if not self.is_payload_valid():
+            raise AuthenticationFailed('Payload not valid')
```

Comment:
Also, I think it's not an `AuthenticationFailed` (that produces a 403) but a `Bad Request` error, that's it, a 400 since there is something data from the request that it's wrong.

Reply:
I don't think we can use a general message here, github call it secret, gitlab token.

Reply:
You can define the message as a class attribute. That's the best. Also, we need a 400 not a 403.